src/createStore.test.ts

```diff
@@ -0,0 +1,54 @@
+import { compressToUTF16 } from 'lz-string';
+import { createStore } from './createStore';
+import { ISavedState, loadStoredState } from './localStorage';
+import { defaultState, IState } from './reducers/states';
+import { history } from './utils/history';
+
+const mockChangedStoredState: ISavedState = {
+  session: {
+    accessToken: 'yep',
+    refreshToken: 'refresherOrb',
+    role: undefined,
+    name: 'Jeff'
+  },
+  playgroundEditorValue: 'Nihao everybody',
+  playgroundIsEditorAutorun: true
+};
+
+const mockChangedState: IState = {
+  ...defaultState,
+  session: {
+    ...defaultState.session,
+    accessToken: 'yep',
+    refreshToken: 'refresherOrb',
+    role: undefined,
+    name: 'Jeff'
+  },
+  workspaces: {
+    ...defaultState.workspaces,
+    playground: {
+      ...defaultState.workspaces.playground,
+      editorValue: 'Nihao everybody',
+      isEditorAutorun: true
+    }
+  }
+};
+
```

Comment:
For `createStore`: you can remove the `else` block and do `let initialStore: IState = defaultState;`

Reply:
I'm not sure how this will work tho? I thought we only want defaultState if loadedStore is null?

Reply:
```
  let initialStore: IState = defaultState;
  if (loadedStore) {
    initialStore = {
      ...defaultState,
      session: {
        ...defaultState.session,
        ...(loadedStore.session ? loadedStore.session : {})
      },
      workspaces: {
        ...defaultState.workspaces,
        playground: {
          ...defaultState.workspaces.playground,
          editorValue: loadedStore.playgroundEditorValue
            ? loadedStore.playgroundEditorValue
            : defaultState.workspaces.playground.editorValue,
          isEditorAutorun: loadedStore.playgroundIsEditorAutorun
            ? loadedStore.playgroundIsEditorAutorun
            : defaultState.workspaces.playground.isEditorAutorun
        }
      }
    };
  } 
```

Reply:
Omg haha thanks